drivers/net/Kconfig

```diff
@@ -72,6 +78,16 @@ config PPP_CLIENT_CLIENTSERVER
 	  This is only necessary if a ppp connection should be
 	  established with a Microsoft Windows PC.
 
+config PPP_DIALUP
```

Comment:
I already use the current PPP implementation as a "dialup" modem, so this is already working (for my use-case). I think introducing a new Kconfig option named "PPP_DIALUP" is misleading, since it is not required for all use-cases. I would prefer instead to add Kconfig options for the missing ppp options, similar to CONFIG_NET_L2_PPP_AUTH_SUPPORT.

Reply:
OK, yes, I see your point, let me still clarify little bit the background of that naming as I was initially also struggling with naming of this config; NET_L2_PPP_SERVER was my initial one. In my head, term "dial up" includes always also setuping/providing IP level access, and PPP_CLIENT_CLIENTSERVER is not providing that for the peer. The changes in this PR are for serving the IP and DNS address, and PPP_DIALUP would need PPP_CLIENT_CLIENTSERVER if working with Windows. I understood that PPP_CLIENT_CLIENTSERVER is needed for triggering the ppp with windows. So, options for new name(s) for the config below, what do you think?
#1: would NET_L2_PPP_SERVER (or PPP_SERVER) be better for include all these changes? 
or
#2: Alternatively if reflecting to these PPP options as you proposed, some options would be: 
CONFIG_NET_L2_PPP_OPTION_SERVE_IP + CONFIG_NET_L2_PPP_OPTION_SERVE_DNS
and
CONFIG_NET_L2_PPP_OPTION_MRU
+ find the way to do things dynamically in net core / sockets side without flaggins (not sure if easily done)

Reply:
Yes, i see that this is not very easy to do, i am not sure about a solution myself. Let me add my comments:

> I understood that PPP_CLIENT_CLIENTSERVER is needed for triggering the ppp with windows

Yes, it is Windows-specific and only ensures that if Windows sends the plain string `CLIENT` to the modem, the modem answers with `CLIENTSERVER`. This is not encapsulated in any protocol and does not provide any further functionality.

> `#1` or `#2`

I would prefer `#2`, even if it is not so easy. I would try to do the following:

1. Add support for `socket(AF_PACKET, SOCK_RAW, IPPROTO_RAW)` to the ip stack, completely independent from ppp. I do not think that any new Kconfig option is needed for that or that the new code need to be guarded by ppp-specific `ifdefs`.
2. Integrate the support for `AF_PACKET` into drivers/net/ppp.c I do not think that this needs to be guarded by any `ifdef` either, it is just a addition to AF_INET and AF_INET6.
3. Add the ppp options from `#2` one by one with dedicated KConfig settings as you suggested.

I would prefer to have as little `ifdefs` in the code as possible (if it makes sense and does not add unnecessary overhead).

Reply:
These are now done. There are now separate configs: NET_L2_PPP_OPTION_MRU, NET_L2_PPP_OPTION_SERVE_IP and NET_L2_PPP_OPTION_SERVE_DNS. Code to support socket(AF_PACKET, SOCK_RAW, IPPROTO_RAW) is now independent from ppp, no ppp flaggings in ip side.